docs/architecture-decisions/support-code-default.md

```diff
@@ -0,0 +1,333 @@
+- Good, because it provides explicit control over default behavior without workarounds
+- Good, because it aligns flagd more closely with OpenFeature specification principles
+- Good, because it supports gradual flag rollout patterns more naturally
+- Good, because it provides the ability to delegate to whatever is defined in code
+- Good, because it requires no changes to existing RPC or protocol signatures
+- Good, because it uses established patterns (field presence) for clear semantics
+- Good, because it maintains full backward compatibility
+- Bad, because it requires updates across multiple components (flagd, providers, testbed)
+- Bad, because it introduces a new concept that users need to understand
+- Neutral, because existing configurations continue to work unchanged
+
+### Remote Evaluation Considerations
+
+This feature works consistently across all flagd evaluation modes through a unified pattern:
+
+1. **In-Process Mode**: Direct evaluation returns responses without value fields when delegating
+2. **RPC Mode (gRPC/HTTP)**: Responses omit the type-specific value field to signal delegation
+3. **OFREP (OpenFeature Remote Evaluation Protocol)**: HTTP responses omit the value field
+
+The key insight is using field omission to indicate "use code default":
+
+- Present value field (even if null/false/empty) = use this value
+- Absent value field = use your code default
+- Works identically across all protocols and evaluation modes
+- No protocol changes required
+
+### Implementation Plan
+
+1. Update flagd-schemas with new JSON schema supporting null default variants
+2. Implement core logic in flagd to handle null defaults and omit value/variant fields
+3. Update flagd-testbed with comprehensive test cases for all evaluation modes
+4. Update OpenFeature providers to handle responses without value fields
+5. Documentation updates, migration guides, and proto comment additions
+
+Note: No protobuf schema changes are required, but implementation must carefully handle field omission
+
+### Testing Considerations
+
+To ensure correct implementation across all components:
+
+1. **Wire Format Tests**: Verify that protobuf messages with omitted fields are correctly serialized without the fields (not with zero values)
+2. **Provider Tests**: Each provider must have tests confirming they detect missing fields correctly in their language
+3. **Integration Tests**: End-to-end tests across different language combinations (e.g., Go flagd with Java provider)
+4. **OFREP Tests**: Verify JSON responses correctly omit fields (not set to null)
+5. **Backward Compatibility Tests**: Ensure old providers handle new responses gracefully
+6. **Consistency Tests**: Verify identical behavior across in-process, RPC, and OFREP modes
+
+### Open questions
+
+- How should providers handle responses with missing value fields in strongly-typed languages?
+- Should we support both `null` and absent `defaultVariant` fields, or choose one approach?
```

Comment:
I think we should support both, I can't see a disadvantage.

Reply:
Yes, exactly.